pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -746,11 +754,11 @@ cdef inline int64_t parse_iso_format_string(str ts) except? -1:
             else:
                 raise ValueError(err_msg)
 
-    if not have_value:
+    if not valid_ts:
```

Comment:
you cannot use have_value or maybe have_value && sign? adding this makes harder to understand

Reply:
`have_value` alone was sufficient